xml/templates/message_templates/contribution_offline_receipt_html.tpl

```diff
@@ -55,78 +55,72 @@
-           {if !empty($getTaxDetails)}
+           {if $isShowTax && '{contribution.tax_amount|raw}' !== '0.00'}
              <th>{ts}Subtotal{/ts}</th>
              <th>{ts}Tax Rate{/ts}</th>
              <th>{ts}Tax Amount{/ts}</th>
            {/if}
            <th>{ts}Total{/ts}</th>
           </tr>
-          {foreach from=$value item=line}
+          {foreach from=$lineItems item=line}
            <tr>
             <td>
-            {if $line.html_type eq 'Text'}{$line.label}{else}{$line.field_title} - {$line.label}{/if} {if $line.description}<div>{$line.description|truncate:30:"..."}</div>{/if}
+              {$line.title}
             </td>
             <td>
              {$line.qty}
             </td>
             <td>
-             {$line.unit_price|crmMoney:$currency}
+             {$line.unit_price|crmMoney:'{contribution.currency}'}
             </td>
-            {if !empty($getTaxDetails)}
+            {if $isShowTax && '{contribution.tax_amount|raw}' !== '0.00'}
               <td>
-                {$line.unit_price*$line.qty|crmMoney:$currency}
+                {$line.unit_price*$line.qty|crmMoney:'{contribution.currency}'}
               </td>
               {if $line.tax_rate || $line.tax_amount != ""}
                 <td>
                   {$line.tax_rate|string_format:"%.2f"}%
                 </td>
                 <td>
-                  {$line.tax_amount|crmMoney:$currency}
+                  {$line.tax_amount|crmMoney:'{contribution.currency}'}
                 </td>
               {else}
                 <td></td>
                 <td></td>
               {/if}
             {/if}
             <td>
-             {$line.line_total+$line.tax_amount|crmMoney:$currency}
+             {$line.line_total+$line.tax_amount|crmMoney:'{contribution.currency}'}
             </td>
            </tr>
           {/foreach}
          </table>
         </td>
        </tr>
-      {/foreach}
+
      {/if}
-     {if !empty($getTaxDetails) && !empty($dataArray)}
+     {if $isShowTax && '{contribution.tax_amount|raw}' !== '0.00'}
        <tr>
          <td {$labelStyle}>
            {ts} Amount before Tax : {/ts}
          </td>
          <td {$valueStyle}>
-           {$formValues.total_amount-$totalTaxAmount|crmMoney:$currency}
+           {$formValues.total_amount-$totalTaxAmount|crmMoney:'{contribution.currency}'}
          </td>
        </tr>
 
-      {foreach from=$dataArray item=value key=priceset}
-        <tr>
-        {if $priceset ||  $priceset == 0 || $value != ''}
-          <td>&nbsp;{$taxTerm} {$priceset|string_format:"%.2f"}%</td>
-          <td>&nbsp;{$value|crmMoney:$currency}</td>
-        {else}
-          <td>&nbsp;{ts}No{/ts} {$taxTerm}</td>
-          <td>&nbsp;{$value|crmMoney:$currency}</td>
-        {/if}
+       {foreach from=$taxRateBreakdown item=taxDetail key=taxRate}
+         <tr>
+          <td>&nbsp;{if $taxRate == 0}{ts}No{/ts} {$taxTerm}{else}{$taxTerm} {$taxDetail.percentage}%{/if}</td>
```

Comment:
I see this use of ts is preexisting. I don't see a quickfix so ok.

Reply:
There's something about the layout that looks off, and the nbsp is also preexisting, but looks wrong. Doesn't need to be in this PR but if the idea is to shortly force an update on existing sites then soon might be a good time to fix it.

Reply:
<img width="551" alt="Untitled2" src="https://user-images.githubusercontent.com/2967821/181522638-7e1fbd19-cf99-4be5-9b4f-82a88d879067.png">